tests/storage/restore.go

```diff
@@ -2156,6 +2069,17 @@ var _ = Describe(SIG("VirtualMachineRestore Tests", func() {
 					Expect(err).ToNot(HaveOccurred())
 					forcedHostAssistedScName = sc.Name
 
+					// Patch the new StorageProfile annotations with the original ones. Needed for the minimumSupportedPvcSize annotation.
+					originalSp, err := virtClient.CdiClient().CdiV1beta1().StorageProfiles().Get(context.Background(), snapshotStorageClass, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if originalSp.Annotations != nil {
```

Comment:
not sure I understand this change

Reply:
The test is using a patched clone of the `storageClass`, but when its new `storageProfile` is created it doesn't copy the original `storageProfile` annotations. This `Patch()` is needed only when the original `storageProfile` was manually annotated with `minimumSupportedPvcSize`. It can be dropped if we want to cover only `storageProfiles` that are auto-annotated with `minimumSupportedPvcSize`.

Reply:
its just not clear what is going on :) what the original storage profile what is the second.. since I understand this code assumes a storageprofile is created in cdi, maybe explain it better? might need to also wait for a SP to be created for the hostAssistedSc and also the code uses the same sc variable for both the original and the hostassisted one which is confusing  (of course its not on you lol) will appreciate if you address this